main.cpp

```diff
@@ -19,13 +19,34 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter &rewriter) : rewriter_(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *castExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        const auto castIgnore = castExpr->getSubExprAsWritten()->IgnoreImpCasts();
+        auto &sourceManager = *Result.SourceManager;
+
+        auto type = Lexer::getSourceText(CharSourceRange::getTokenRange(
+                                            castExpr->getLParenLoc().getLocWithOffset(1),
+                                            castExpr->getRParenLoc().getLocWithOffset(-1)),
+                                            sourceManager, Result.Context->getLangOpts());
+											
+        auto replaceRange = CharSourceRange::getCharRange(castExpr->getLParenLoc(), 
+                                                          castExpr->getSubExprAsWritten()->getBeginLoc());

```

Comment:
Please describe the difference between `getSubExprAsWritten()->getBeginLoc()` and `getRParenLoc().getLocWithOffset(1)`.

Reply:
If we have source code:
```
int main() {
    float xyz;
    int i = (int)xyz;
    return 0;
}
```

1. `getSubExprAsWritten()->getBeginLoc()` returns the starting position of the given subexpression, as it was written in the source.  

For this code it is **‘x’.**

2. `getRParenLoc().getLocWithOffset(1)`: 
- `getRParenLoc()` returns the position of the closing parenthesis
- `getLocWithOffset(1)` return a source location with the specified offset, in our case, the offset is 1. 

For this code it is **‘x’.**

So we get the same result. It turns out that `getSubExprAsWritten() -> getBeginLoc()` and `getRParenLoc().getLocWithOffset(1)` is equivalent, at least for this example.

Reply:
Just for this result. The difference can be seen for the following example:
```cpp
int main() {
    float xyz;
    int i = (int) xyz;
    return 0;
}
```

Reply:
For this example result `getSubExprAsWritten()->getBeginLoc()` is:
```с
int main() {
    float xyz;
    int i = static_cast<int>(xyz);
    return 0;
}
```
Result `getRParenLoc().getLocWithOffset(1)` is:
```
int main() {
    float xyz;
    int i = static_cast<int>( xyz);
    return 0;
}
```